src/main/java/frc/team997/lib/trajectory/TrapezoidProfile.java

```diff
@@ -0,0 +1,114 @@
+/**
+Copyright 2022 FRC Team 997
+
+This program is free software: 
+you can redistribute it and/or modify it under the terms of the 
+GNU General Public License as published by the Free Software Foundation, 
+either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, 
+but WITHOUT ANY WARRANTY; without even the implied warranty of 
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with SpartanLib2. 
+If not, see <https://www.gnu.org/licenses/>.
+*/
+package frc.team997.lib.trajectory;
+
+import java.util.Objects;
+
+/**
+ * Represents a special case of an {@link AsymmetricTrapezoidProfile} where the maxAcceleration and
+ * maxDeceleration constraints are the same.
+ *
+ * <p>A motion profile that allows for smooth motion between two states. The dx/dt of this motion
+ * (velocity) forms the trapezoid shape on a graph. The particular elements of motion are velocity
+ * and acceleration limited.
+ *
+ * <p>The most useful practical application of this is use defining the setpoint for a PID (or
+ * similar) controller, to avoid control effort saturation or the initial spike in input from the P
+ * term.
+ *
+ * <p>Also could be used to meet physical constraints from a mechanism (can't go too fast or the
+ * wires might tangle, mechanism has too much mass to decelerate too quickly, etc...).
+ */
+public class TrapezoidProfile extends AsymmetricTrapezoidProfile {
+
+    /** Data class to hold the maximum allowed rates for the output of a TrapezoidProfile. */
+    public static class TrapezoidProfileConstraints {
```

Comment:
rename this constraints too. if theres a problem in the tests file both importing constraints, remove the import for constraints and always reference as TrapezoidProfile.Constraints